File: LeetCode/python/FindMinimumInRotatedSortedArrayII.py
class Solution(object):
    def findMin(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        preNum = nums[0]
        numsSize = len(nums)
        for i in range(1,numsSize,1):
            if(nums[i] >= preNum):
                preNUm = nums[i]
            else:
                return nums[i]
                
        return nums[0]    
			
#测试程序
s = Solution()
nums = [0,1,2,3,4,5,6,7]
print(s.findMin(nums))

File: LeetCode/iter1/c++/FindMinimumInRotatedSortedArrayII.cpp
#include <iostream>
#include <vector>

using namespace std;

int findMin(vector<int>& nums) {
	int preNum = nums[0];
	
	for(int i=1;i<nums.size();i++)
	{
		if(nums[i] >= preNum)
			preNum = nums[i];
		else
			return nums[i];			
	}
	
	return nums[0];        
}

int main()
{
	vector<int> v;
	v.push_back(3);
	v.push_back(3);
	v.push_back(4);
	v.push_back(4);
	v.push_back(5);
	v.push_back(5);	
	v.push_back(2);
	v.push_back(2);

	cout<<findMin(v);
	
	return 0;
}